api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirImmunizationServiceImplTest.java

```diff
@@ -188,6 +237,43 @@ public void updateImmunization_shouldUpdateImmunizationAccordingly() {
 		assertThat(members.get(CIEL_165907).getValueDate(), equalTo(new DateType("2020-10-08").getValue()));
 	}
 	
+	@Test
+	public void updateImmunization_shouldUpdateNoteFieldWhenNoteConceptIsAvailable() throws Exception {
+		FhirContext ctx = FhirContext.forR4();
+		IParser parser = ctx.newJsonParser();
+		String createJson = IOUtils.toString(
+		    Objects.requireNonNull(
+		        getClass().getResourceAsStream("/org/openmrs/module/fhir2/providers/immunization-note.json")),
+		    StandardCharsets.UTF_8);
+		Immunization created = service.create(parser.parseResource(Immunization.class, createJson));
+		assertThat(created.getNoteFirstRep().getText(), is("This is a test immunization note."));
+		
+		Immunization immunizationToBeUpdated = service.get(created.getId());
+		immunizationToBeUpdated.getNote().clear();
+		immunizationToBeUpdated.addNote().setText("This is an UPDATED immunization note.");
+		Immunization updated = service.update(created.getId(), immunizationToBeUpdated);
+		assertThat(updated.getNoteFirstRep().getText(), is("This is an UPDATED immunization note."));
+	}
+	
+	@Test
+	public void updateImmunization_shouldNotFailIfNoteConceptIsMissingButNoteIsProvided() throws Exception {
```

Comment:
So if we do not have the note concept but a note is provided, do we just silently throw away the provided note?

Reply:
This is pretty much the standard behaviour for FHIR; if you don't know how to process a field or property, it is dropped. On each FHIR operation, the server returns it's representation of the resource so a client can compare what it sent to what it got back to see which fields were not supported.